lua/terrortown/menus/gamemode/base_gamemodemenu.lua

```diff
@@ -102,6 +107,56 @@ function CLGAMEMODEMENU:AddSubmenu(submenu)
 	self.submenus[#self.submenus + 1] = submenu
 end
 
+---
+-- Enables searchbar for the menu.
+-- @param bool active
+-- @realm client
+function CLGAMEMODEMENU:EnableSearchbar(active)
+	self.searchbar = active
+end
+
+---
+-- Checks if the menu has a searchbar enabled.
+-- @return boolean Returns true if searchbar is set
+-- @realm client
+function CLGAMEMODEMENU:HasSearchbar()
+	return self.searchbar
+end
```

Comment:
In my opinion I'd rewrite this. All other settings are based on baseclass overwrites, not on variables. So just a `HasSearchbar()` that can be overwritten should be used imho